### ISSUE TITLE
[LibreOffice add-on] Crashed with latest nightly - 2022-08-30

### ISSUE THREAD

**Issue by marcoagpinto** [languagetool-org/languagetool#7054]
Hello @FredKruse 

I was revising my thesis:
![LT_crash_20220830](https://user-images.githubusercontent.com/5192600/187538527-c767b799-1195-4587-a65a-c563c28c91a5.png)

**Comment by marcoagpinto** [comment]
Basically, I opened the .docx, changed all text language to Portuguese, then selected “Save As” and when I was selecting .ODT format, the window above popped-up.

**Comment by FredKruse** [comment]
@marcoagpinto Your thesis seams to be a hardness test for the extension. :smile: 
Could you please reproduce the issue, close the document and send me the log file right after?
But, I can look at it, not before the beginning of next week.
Note: The check should run after closing the popup, but not all text level rules will be handled correct.

**Comment by marcoagpinto** [comment]
@FredKruse 

If it crashes again, I will send you a log file.

How do I get this log file?

Thanks!

**Comment by marcoagpinto** [comment]
Better yet, after the crash I only used LibreOffice a few times and on the thesis, so I can send you the log I have?

Please tell me where the log is for me to get it.

Thanks!

**Comment by FredKruse** [comment]
The log file is written new every time you start LibreOffice again. So, I need the log-file from the session in which the crash appears.
The name of the log-file is: LanguageTool.log
The directory-path is: \\user\<YourUserName>\AppData\Roaming\languagetool.org\LanguageTool\LibreOffice

**Comment by marcoagpinto** [comment]
@FredKruse 

It crashed again:
![lt_crash_20220831](https://user-images.githubusercontent.com/5192600/187745233-b0d07197-16aa-4a6b-91ba-109e902d99ae.png)

Same procedure causing it:
1. Open .docx;
2. Change all text language to Portuguese;
3. Save as;
4. Select extension .ODT
5. ... and the crash happens.

Regarding the log file it is 3 MB long, and it reveals tons of information (contents) of the thesis, and therefore I can't share it here in a public place.

**Comment by marcoagpinto** [comment]
28878 lines of log.

**Comment by marcoagpinto** [comment]
And it shows full paragraphs of the thesis, I can't share it here.

**Comment by marcoagpinto** [comment]
Here is also a hang (LibreOffice stopped responding and I had to close it)
[LanguageTool_LibreOffice_hang_20220902.log](https://github.com/languagetool-org/languagetool/files/9473918/LanguageTool_LibreOffice_hang_20220902.log)

**Comment by marcoagpinto** [comment]
@FredKruse @danielnaber 

I know I shouldn't share this, my master's dissertation from ten years ago, but it is ten years old and so, it really doesn't matter.

Open the dissertation (.doc) with LibreOffice, go to Tools and change the whole language of the document to “Portuguese”, then go to the menu option to save as to select .ODT and voilà, the add-on crashes.

Here is the .doc + the log of the crash.

It is good that it also crashes with this .doc, so I don't have to share the PhD thesis, which is classified.

[dissertation_and_crash_log_20220903.zip](https://github.com/languagetool-org/languagetool/files/9481796/dissertation_and_crash_log_20220903.zip)

**Comment by FredKruse** [comment]
@marcoagpinto Thank you for providing your document. I found two different problems. The issue should be solved.
Please test it with your PhD thesis. 
I will remove the document from my personal machine.

**Comment by marcoagpinto** [comment]
@FredKruse 

Thank you, my dear brother!

❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by marcoagpinto** [comment]
@FredKruse 

Look at this:
![lt_crash_20220906](https://user-images.githubusercontent.com/5192600/188710243-75170ed5-9ef4-455d-92cb-771f5630d5b3.png)

Exactly the same procedure:
1. Open .docx file;
2. Go to the menu to change the whole text language of the document to Portuguese;
3. Save as to select .odt format;
4. This window pops up.

**Comment by FredKruse** [comment]
@marcoagpinto That is frustrating. Is it the test with the PhD thesis?
Could you test, If the same problem accure, if you save your dissertation as docx (MS-Office) and open it again with LibreOffice doing the same steps you described?
I need a document to test the problem.

**Comment by FredKruse** [comment]
An other idea: The problem seams to be at the beginning of the text paragraphs. Could you generate a document of the First two or three pages of document as docx. Pleased Test, If the same message appears. Maybe you could send me this document for testing.

**Comment by marcoagpinto** [comment]
@FredKruse 

Does this video help?



https://user-images.githubusercontent.com/5192600/188732802-c0132c74-80de-4d21-b25a-eaa10e51007d.mp4

Thanks!

**Comment by marcoagpinto** [comment]
@FredKruse 

Hello!

Still with my master’s dissertation.

I pressed CTR + A and pasted the whole text at the end some 4 or 5 times until it became 576 pages longs.

Then changed the language to Portuguese (but it was already Portuguese).

I was looking at the logs, and it has many nulls.

I am uncertain if this helps, but here is the 576 pages odt + log.

Please tell me if this helps.

Thanks!

[dissertation_and_crash_log_20220907.zip](https://github.com/languagetool-org/languagetool/files/9502662/dissertation_and_crash_log_20220907.zip)

**Comment by FredKruse** [comment]
@marcoagpinto Thank you for your test and the document.
Unfortunately, the video and your master dissertation didn't help. The reason of the problem is located in the structure of the PhD document.
Fortunately, I was able to generate a test document, that shows a very similar error, which I have solved. Please test, if it solves the problem with your PhD thesis too.
I also introduced a repair mechanism for not correctly mapped text paragraphs. Please look at the log file, if no error message appears. Send me your log file if there are any warnings inside.
I am on private business until the end of this week. So, I can work on LO extension at the beginning of next week.

**Comment by marcoagpinto** [comment]
Thank you!

❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by marcoagpinto** [comment]
@FredKruse 

Well,

![lt_crash_20220907](https://user-images.githubusercontent.com/5192600/188952481-7e373eba-9043-491f-979d-06001b1ba65a.png)

The log has almost my entire thesis, and it complains about all the headings (chapters, subchapters).

I can't share it, the only part which shouldn't be confidential is this:

```
LT office integration logs from Wed Sep 07 19:27:12 BST 2022
LanguageTool 5.9-SNAPSHOT (2022-09-07 16:42:34 +0000, b9af269)
OS: Windows 10 10.0 on amd64
LibreOffice 7.4.0.3 (The Document Foundation), en-GB
Java-Version: 1.8.0_341, max. Heap-Space: 7262 MB, LT Heap Space Limit: 6536 MB

getLTDictionaryFile: LT dictionary file doesn't exist: start to create
Caches read from: C:\Users\marco\AppData\Roaming\languagetool.org\LanguageTool\LibreOffice\cache\LtCache1.lcz
Version or active rules have changed: Cache rejected (Cache Version: 5.9-SNAPSHOT, actual LT Version: 5.9-SNAPSHOT)
MultiDocumentsHandler: getNumDoc: Document 0 created; docID = 1
Time to generate LT spelling file: 372
setLtDictionary: Create LT spelling dictionary __LT_pt-PT_internal1.dic
Internal LT dicitionary for language pt-PT added: Number of words = 12562
Time to generate dictionary: 64
WARNING: DocumentCache: Could not map Paragraph(23): '  V1.0'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(24): '"If you know the enemy and know yourself, you need not fear the result of a hundred battles.'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(26): ' If you know yourself but not the enemy, for every victory gained you will also suffer a defeat.'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(28): ' If you know neither the enemy nor yourself, you will succumb in every battle."'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(29): '       			— Sun Tzu, The Art of War'; secondTextDone: true

```

And, look at this:
```
WARNING: DocumentCache: Could not map Paragraph(8309): '1'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8310): '2'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8311): '3'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8312): '4'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8313): '5'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8314): '6'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8315): '7'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8316): '8'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8317): '9'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8318): '10'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8319): '11'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8320): '12'; secondTextDone: true
WARNING: DocumentCache: Could not map Paragraph(8321): 'Esta página foi intencionalmente deixada em branco.'; secondTextDone: true
```
I separate each chapter of the thesis with 12 lines, each one with a number (1,2,3,4,5 until 12) and then write the text that the page was left blank on purpose.

The 12 lines were set to white colour to not appear visible, and after the "page left blank blah blah" I have probably some empty lines and a manual page break.

**Comment by marcoagpinto** [comment]
EDIT: The "V1.0" and Sun Tzu quotations are two different text gadgets.

I have text gadgets mixed with ordinary text.

**Comment by FredKruse** [comment]
I can work at this problem not until next monday.
But I need a test document.
Here is the idea:
Please open your PhD thesis with the application you created it. Delete all pages from page 4 to the end. The both gadgets should be inside the left text. You can change the ordinary text to bla bla.  But the structure should be the same.
Save the document to something like testdoc.docx.
Close all and open it again with LibreOffice. Do the steps you have done with your PhD thesis.
When it crashs, we have a test document and you can left it here.

**Comment by marcoagpinto** [comment]
@FredKruse 

Cool, we now have a plan.

Before Monday, you will have a test document.

❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by marcoagpinto** [comment]
@FredKruse 

I was removing text from the thesis and checking the log file, and it had tons of text.

Then I found out that the log doesn't get deleted every time I start LibreOffice, so I will have to start from zero and delete the log before opening LibreOffice.

Here is an example of the log repeating itself:
[LanguageTool_old.log](https://github.com/languagetool-org/languagetool/files/9525760/LanguageTool_old.log)

**Comment by marcoagpinto** [comment]
@FredKruse 

I started from zero.

I deleted the log file and opened the original .docx, changed all language to Portuguese and saved as .ODT.

LanguageTool crashed, I closed the crash window and let the add-on continue checking the thesis.

I had the log file folder open and its size kept increasing and high CPU load on LibreOffice.

When it reached 50 MB I closed LibreOffice.

The log repeated the sentence:
`java.lang.ArrayIndexOutOfBoundsException`

![repeated_line_in_log_file_20220908](https://user-images.githubusercontent.com/5192600/189096206-52b92eb3-c608-4ba0-9b05-76ddb417d02f.png)

That line was repeated over a million times in the log.

Until something is done to fix this and the issue of my previous comment where the log isn't deleted, I will stop testing the thesis.
